docs/walkthroughs/running-with-secure-hdfs.md

```diff
@@ -0,0 +1,115 @@
+# Fully secure HDFS and Spark with Kerberos, SASL, and TLS. 
+This walkthrough will show how to run a version of the [TeraSort Benchmark](https://github.com/ehiggs/spark-terasort) with Spark in a fully secure deployment and Kerberos-secured HDFS as the storage backend. Specifically the following:
+1.  Use a Kerberos-secured HDFS with Spark.
+1.  Use TLS to secure the file-serving channels of your Spark jobs.
+1.  Use SASL for authentication of Spark Executors and encryption of the BlockTransferService
+
+Spark can use [Kerberos](https://en.wikipedia.org/wiki/Kerberos_(protocol)) to authenticate and provide SASL with HDFS running on DC/OS. To start this walkthrough we assume that you have the following:
+1.  DC/OS Cluster with 5 agents with >= 4 cores/agent.
+1.  A running KDC that you can add principals to. 
+1.  A way to download/acquire the `keytab` from the KDC to your local workstation.
+
+## Setting up secure HDFS
+The Keberos `keytab` is a binary file and cannot be uploaded to the Secret Store directly. To use binary secrets in DC/OS 1.10 and 1.11 the binary file must be base64 encoded and the resultant string will be uploaded with the prefix: `__dcos_base64__<secretname>`, this tells DC/OS to decode the file before placing it in the Sandbox.
+
+1.  Establishing correct principals for HDFS (assumes you're using the HDFS from the DC/OS Universe and installed with service name `hdfs`). **Note** Requires DC/OS EE for file-based secrets. Add the following principals to the KDC (Obviously it is recommended to use a script or other automation to establish these principals):
```

Comment:
~~Establishing~~ -> Establish  (to be in the same command form as the following bullets)